### PR TITLE
dask-cuda v22.4.0

### PR DESCRIPTION
**Pull request by charlesbluca**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Also changes the Python / OS constraints - this would be done to match up with the behavior of the RAPIDS dask-cuda 22.02 packages, which currently only support Linux and python 3.8, 3.9.

### CODE REVIEW

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Comment by charlesbluca** [comment]
Failures seem to be related to click - we might want to set a constraint to prevent pulling in 8.1.0

**Comment by jakirkham** [comment]
The python version is coming from [the `setup.py`]( https://github.com/rapidsai/dask-cuda/blob/6ca650ccbe5e20c101c1881d6cd1b0b3f736f0c1/setup.py#L41 ). Can we please update PR ( https://github.com/rapidsai/dask-cuda/pull/883 ) to include that?

**Comment by jakirkham** [comment]
Yeah Distributed fixes that in PR ( https://github.com/dask/distributed/pull/6018 ), which will show up in the new release. Though that won't help us until we can bump the version constraints

We can add a version constraint to `click` here in `run`. Though we may also need to hot-fix old versions to add that constraint ( similar to PR https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/204 )

**Comment by charlesbluca** [comment]
Cool - if we're opening a hotfix PR, can / should we also add `__unix` as a runtime dep for the older packages?

**Comment by jakirkham** [comment]
Yeah that makes sense

We should probably also add the `click` constraint to `dask-cuda` ( https://github.com/rapidsai/dask-cuda/pull/883 )

**Comment by jakirkham** [comment]
Since these changes are also going to be part of the new `dask-cuda` release, thinking it may make sense to roll this into the release update (bumping the version here). Thoughts? 🙂

**Comment by charlesbluca** [comment]
Sure! I'm happy to do that 🙂

**Comment by charlesbluca** [comment]
In that case, do we want to hold off on merging this PR until a new dask-cuda version is published to PyPI?

**Comment by jakirkham** [comment]
Yeah that makes sense. Thanks for getting the ball rolling here Charles! 😄

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ requirements:
     - numba >=0.53.1
     - numpy >=1.16.0
     - pynvml >=11.0.0
-    - click <8.1
+    - click =8.0.4
```
Comment (jakirkham) [inline recipe/meta.yaml]: ~~We discussed offline and decided to match what was done in PR ( https://github.com/rapidsai/integration/pull/453 ). The `distributed` issue this stems from is a bit different than the one that `black` encountered, but this pinning should work just as well here~~

Edit: Woops wrong PR 😅 Moved comment here ( https://github.com/rapidsai/dask-cuda/pull/883#discussion_r838852573 ).

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,23 +11,25 @@ source:
...
   run:
-    - python >=3.7
+    - python >=3.8,<3.10
     - dask >=2021.11.1,<=2022.01.0
     - distributed >=2021.11.1,<=2022.01.0
     - numba >=0.53.1
     - numpy >=1.16.0
     - pynvml >=11.0.0
+    - click =8.0.4
+    - __unix
```
Comment (jakirkham) [inline recipe/meta.yaml]: Think about this more as `__unix` includes macOS, we might want to go with `__linux` to restrict it further.

```suggestion
    - __linux
```

**Comment by jakirkham** [comment]
Could you please update this to the latest version? Can incorporate PR ( https://github.com/conda-forge/dask-cuda-feedstock/pull/10 ) if it helps

**Comment by charlesbluca** [comment]
@jakirkham done

File: recipe/meta.yaml [hunk 3]
```diff
@@ -19,15 +19,17 @@ build:
...
 requirements:
   host:
-    - python >=3.7
+    - python >=3.8,<3.10
     - pip
   run:
-    - python >=3.7
+    - python >=3.8,<3.10
     - dask >=2021.11.1,<=2022.01.0
     - distributed >=2021.11.1,<=2022.01.0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Think we need to [bump these]( https://github.com/rapidsai/dask-cuda/blob/451b3b3a5fb5aa2a194d8acbb966ccaaa020fff1/requirements.txt#L1-L2 ) too

```suggestion
    - dask==2022.03.0
    - distributed==2022.03.0
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -19,15 +19,17 @@ build:
...
-    - python >=3.7
+    - python >=3.8,<3.10
     - pip
   run:
-    - python >=3.7
-    - dask >=2021.11.1,<=2022.01.0
-    - distributed >=2021.11.1,<=2022.01.0
+    - python >=3.8,<3.10
+    - dask==2022.03.0
+    - distributed==2022.03.0
```
Comment (charlesbluca) [inline recipe/meta.yaml]:
```suggestion
    - dask ==2022.03.0
    - distributed ==2022.03.0
```

Comment (charlesbluca) [reply recipe/meta.yaml]:
```suggestion
    - dask =2022.03.0
    - distributed =2022.03.0
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -19,15 +19,17 @@ build:
...
-    - python >=3.7
+    - python >=3.8,<3.10
     - pip
   run:
-    - python >=3.7
-    - dask >=2021.11.1,<=2022.01.0
-    - distributed >=2021.11.1,<=2022.01.0
+    - python >=3.8,<3.10
+    - dask =2022.03.0
+    - distributed =2022.03.0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Sorry forgot about the space, but think we still want `==`

```suggestion
    - dask ==2022.03.0
    - distributed ==2022.03.0
```

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Pull request merged**